spring-cloud-aws-appconfig-config/src/main/java/org/springframework/cloud/aws/appconfig/AwsAppConfigProperties.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.aws.appconfig;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.util.StringUtils;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+/**
+ * @author jarpz
+ */
+@ConfigurationProperties(AwsAppConfigProperties.CONFIG_PREFIX)
+public class AwsAppConfigProperties implements Validator {
```

Comment:
I can move this class to autoconfigure module, but I need it to create `GetConfigurationRequest` inside `AwsAppConfigPropertySource` and also `AwsAppConfigPropertySourceLocator`  code.. How would do it in this case?

Reply:
Those classes should receive raw properties in order to make them work via set method maybe instead of inject the properties in the constructor with `AwsAppConfigProperties configProperties`.